modules/passkey/test/userstories/OffchainPasskeyVerification.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { buildSafeTransaction } from '../utils/safe'
+
+describe('Offchain Passkey Signature Verification [@userstory]', () => {
+  const setupTests = deployments.createFixture(async ({ deployments }) => {
+    const { SafeProxyFactory, SafeL2, FCLP256Verifier, WebAuthnSignerFactory } = await deployments.run()
+
+    const proxyFactory = await ethers.getContractAt(SafeProxyFactory.abi, SafeProxyFactory.address)
+    const singleton = await ethers.getContractAt(SafeL2.abi, SafeL2.address)
+    const verifier = await ethers.getContractAt('IP256Verifier', FCLP256Verifier.address)
+    const signerFactory = await ethers.getContractAt('WebAuthnSignerFactory', WebAuthnSignerFactory.address)
+
+    const navigator = {
+      credentials: new WebAuthnCredentials(),
+    }
+
+    return {
+      proxyFactory,
+      singleton,
+      signerFactory,
+      navigator,
+      verifier,
+      SafeL2,
+    }
+  })
+
+  it('should be possible to verify offchain passkey signature', async () => {
+    const { proxyFactory, singleton, signerFactory, navigator, verifier, SafeL2 } = await setupTests()
+
+    const verifierAddress = await verifier.getAddress()
+
+    // Creating the credentials for Passkey.
+    const credential = navigator.credentials.create({
+      publicKey: {
+        rp: {
+          name: 'Safe',
+          id: 'safe.global',
+        },
+        user: {
+          id: ethers.getBytes(ethers.id('chucknorris')),
+          name: 'chucknorris',
+          displayName: 'Chuck Norris',
+        },
+        challenge: ethers.toBeArray(Date.now()),
+        pubKeyCredParams: [{ type: 'public-key', alg: -7 }],
+      },
+    })
+
+    // Getting the publicKey from the credential and creating the signer.
+    const publicKey = decodePublicKey(credential.response)
+    await signerFactory.createSigner(publicKey.x, publicKey.y, verifierAddress)
+
+    // Querying the signer.
+    const signerAddress = await signerFactory.getSigner(publicKey.x, publicKey.y, verifierAddress)
+    const signer = await ethers.getContractAt('WebAuthnSigner', signerAddress)
+
+    // Deploying Safe with the signer as a Single Owner.
+    const singletonAddress = await singleton.getAddress()
+    const setupData = singleton.interface.encodeFunctionData('setup', [
+      [signerAddress],
+      1,
+      ethers.ZeroAddress,
+      '0x',
+      ethers.ZeroAddress,
+      ethers.ZeroAddress,
+      0,
+      ethers.ZeroAddress,
+    ])
+    const safeAddress = await proxyFactory.createProxyWithNonce.staticCall(singletonAddress, setupData, 0)
+    await proxyFactory.createProxyWithNonce(singletonAddress, setupData, 0).then((tx: any) => tx.wait())
+    const safe = await ethers.getContractAt(SafeL2.abi, safeAddress)
+
+    // Calling `checkSignatures(...)` (With passkey signed dataHash - EIP1271) on the created Safe and check it doesn't revert.
+    const nonce = await safe.nonce()
```

Comment:
While this does kind of similar, we should be using EIP-1271 `isValidSignature` on the Safe and not checking signatures for a Safe transaction as per the user story.

Reply:
Did you mean with this one? https://github.com/safe-global/safe-smart-account/blob/v1.4.1-build.0/contracts/handler/CompatibilityFallbackHandler.sol#L28

Reply:
Done 👍🏾 Thanks for the help @nlordell